### PR TITLE
fix nuget package related warnings 

### LINKED ISSUE #985: Some warning and issues
Hello, 
I'm using using VS 2017 15.9.10 and getting some warnings and errors.
1. Getting errors while building the project, why do we need to run build task for library manager since all the required files are already on the repo (#848)
2. Compiler gives warning about **Razor syntax** for rendering partial pages to be changed to Tag Helpers
3. The NuGet package version conflict: Microsoft.ApplicationInsights
4. Using version in Microsoft.AspNetCore.App
5. Called some deprecated methods in authorize filters & obsolete methods in logging Adding ApplicationInsights in Startup
6. There are some async/await warnings

### PR DESCRIPTION
**Pull request by rafsanulhasan**
fix #985

### CODE REVIEW

File: src/Web/WebMVC/libman.json [hunk 1]
```diff
@@ -9,14 +9,6 @@
     {
       "provider": "unpkg",
       "library": "bootstrap@4.3.1",
-      "files": [
```
Comment (mvelosop) [inline src/Web/WebMVC/libman.json]: These "files change" is included in another PR and don't seem necessary, for can't reproduce the error. See PR https://github.com/dotnet-architecture/eShopOnContainers/pull/994#issuecomment-479929262

File: src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj [hunk 2]
```diff
@@ -17,8 +17,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj]: Why include references that are not needed?

File: src/Services/Basket/Basket.UnitTests/Basket.UnitTests.csproj [hunk 3]
```diff
@@ -7,7 +7,11 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Basket/Basket.UnitTests/Basket.UnitTests.csproj]: Why include references that are not needed?

Comment (rafsanulhasan) [reply src/Services/Basket/Basket.UnitTests/Basket.UnitTests.csproj]: removed!

File: src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj [hunk 4]
```diff
@@ -17,8 +17,14 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Services/Basket/Basket.UnitTests/Basket.UnitTests.csproj [hunk 5]
```diff
@@ -7,7 +7,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Basket/Basket.UnitTests/Basket.UnitTests.csproj]: Please remove unneeded packages

File: src/Services/Catalog/Catalog.FunctionalTests/Catalog.FunctionalTests.csproj [hunk 6]
```diff
@@ -33,8 +33,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Catalog/Catalog.FunctionalTests/Catalog.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Services/Catalog/Catalog.UnitTests/Catalog.UnitTests.csproj [hunk 7]
```diff
@@ -7,7 +7,11 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Catalog/Catalog.UnitTests/Catalog.UnitTests.csproj]: Please remove unneeded packages

File: src/Services/Location/Locations.FunctionalTests/Locations.FunctionalTests.csproj [hunk 8]
```diff
@@ -17,8 +17,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Location/Locations.FunctionalTests/Locations.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Services/Marketing/Marketing.FunctionalTests/Marketing.FunctionalTests.csproj [hunk 9]
```diff
@@ -17,8 +17,13 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Marketing/Marketing.FunctionalTests/Marketing.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Services/Ordering/Ordering.FunctionalTests/Ordering.FunctionalTests.csproj [hunk 10]
```diff
@@ -17,8 +17,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Ordering/Ordering.FunctionalTests/Ordering.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Services/Ordering/Ordering.UnitTests/Ordering.UnitTests.csproj [hunk 11]
```diff
@@ -7,7 +7,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.2.0" />
```
Comment (mvelosop) [inline src/Services/Ordering/Ordering.UnitTests/Ordering.UnitTests.csproj]: Please remove unneeded packages

File: test/ServicesTests/Application.FunctionalTests/Application.FunctionalTests.csproj [hunk 12]
```diff
@@ -67,8 +67,14 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.2.0" />
```
Comment (mvelosop) [inline test/ServicesTests/Application.FunctionalTests/Application.FunctionalTests.csproj]: Please remove unneeded packages

File: src/Web/WebMVC/libman.json [hunk 13]
```diff
@@ -26,9 +26,9 @@
     {
       "library": "jquery-validate@1.19.0",
       "destination": "wwwroot/lib/jquery-validate/",
-      "files": [
-        "jquery.validate.min.js",
-        "jquery.validate.js"
+      "files": [
```
Comment (mvelosop) [inline src/Web/WebMVC/libman.json]: This change just adds trailing spaces, I don't usually care about this, but perhaps just revert the changes in this file!

File: src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj (hunk 4 above)
Comment (rafsanulhasan) [reply src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj]: INHO, these are need to resolve version conflicts

Comment (mvelosop) [reply src/Services/Basket/Basket.FunctionalTests/Basket.FunctionalTests.csproj]: They are not.

These are the conflicts both with or without these references:

```txt
Basket.FunctionalTests.csproj : warning NU1608: Detected package version outside of dependency constraint: Microsoft.ApplicationInsights.PerfCounterCollector 2.5.1 requires Microsoft.ApplicationInsights (= 2.5.1) but version Microsoft.ApplicationInsights 2.6.1 was resolved.
WebMVC.csproj : warning NU1608: Detected package version outside of dependency constraint: Microsoft.ApplicationInsights.PerfCounterCollector 2.5.1 requires Microsoft.ApplicationInsights (= 2.5.1) but version Microsoft.ApplicationInsights 2.6.1 was resolved.
Basket.API.csproj : warning NU1608: Detected package version outside of dependency constraint: Microsoft.ApplicationInsights.PerfCounterCollector 2.5.1 requires Microsoft.ApplicationInsights (= 2.5.1) but version Microsoft.ApplicationInsights 2.6.1 was resolved.
Locations.API.csproj : warning NU1608: Detected package version outside of dependency constraint: Microsoft.ApplicationInsights.PerfCounterCollector 2.5.1 requires Microsoft.ApplicationInsights (= 2.5.1) but version Microsoft.ApplicationInsights 2.6.1 was resolved.
```

No real surprises here, since conflicts are related to AI, changes in non-AI-related packages should make no difference.

**Comment by escalonn** [comment]
This PR is addressing not only the AI package warnings, but also the errors/warnings resulting from not referencing `AspNetCore.App` in the test projects, as described [here](https://github.com/aspnet/AspNetCore/issues/3156). It's not going about it the right way, though, it's meant to be done with a reference to the meta-package and not to individual packages.

**Comment by mvelosop** [comment]
Thanks @escalonn, I just realized that now we are even getting compilations errors. I'm guessing it's because of the version removal from the metapackages, but that's the way it has to be.

Just a few days ago we were discussing this on PR #999, and PR #1000 fixed it.

The solution is documented here: https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.2#test-app-prerequisites

I'm fixing the compilation errors on unit tests right now, and investigating on a couple of failing tests.
